osu.Game.Rulesets.Taiko/Mods/TaikoModClassic.cs

```diff
@@ -12,16 +17,98 @@ namespace osu.Game.Rulesets.Taiko.Mods
 {
     public class TaikoModClassic : ModClassic, IApplicableToDrawableRuleset<TaikoHitObject>, IApplicableToDrawableHitObject
     {
+        private IReadOnlyList<Mod> mods = Array.Empty<Mod>();
+
+        private const float hd_base_fade_out_duration = 0.375f;
+
+        private const float hd_base_initial_alpha = 0.75f;
+
+        private const float hdhr_base_fade_out_duration = 0.2f;
+
+        private const float hdhr_base_initial_alpha = 0.2f;
+
+        private const float hidden_base_aspect = 4f / 3f;
+
+        private readonly BindableFloat hiddenFadeOutDuration = new BindableFloat(hd_base_fade_out_duration);
+
+        private readonly BindableFloat hiddenInitialAlpha = new BindableFloat(hd_base_initial_alpha);
+
         public void ApplyToDrawableRuleset(DrawableRuleset<TaikoHitObject> drawableRuleset)
         {
             var drawableTaikoRuleset = (DrawableTaikoRuleset)drawableRuleset;
-            drawableTaikoRuleset.LockPlayfieldAspectRange.Value = false;
+
+            // drawableRuleset.Mods should always be non-null here, but just in case.
+            mods = drawableRuleset.Mods ?? mods;
+
+            drawableTaikoRuleset.MaximumAspect.Value = 22f / 9f;
+            drawableTaikoRuleset.MinimumAspect.Value = 5f / 4f;
+            drawableTaikoRuleset.TrimOnOverflow.Value = true;
+
+            TaikoModHidden? hidden = mods.OfType<TaikoModHidden>().FirstOrDefault();
+
+            if (mods.OfType<TaikoModHardRock>().Any())
+            {
+                // For hardrock, the playfield time range is clamped to within classicMaxTimeRange and the equivalent
+                // time range for a 16:10 aspect ratio.
+                drawableTaikoRuleset.TrimOnOverflow.Value = false;
+
+                // Apply stable aspect ratio limits for hardrock (visually taken)
+                drawableTaikoRuleset.MaximumAspect.Value = 1.963f;
+
+                // This is accurate to 4:3, but slightly off for 5:4
+                drawableTaikoRuleset.MinimumAspect.Value = 1.666f;
+
+                drawableTaikoRuleset.MinimumRelativeHeight.Value = 0.26f;
+                drawableTaikoRuleset.MaximumRelativeHeight.Value = 0.26f;
```

Comment:
Why does this have to be configurable / adjusted at all?

Reply:
This is to match stable's scaling, without it the playfield sizing scales differently from stable

Reply:
"scales differently" is kind of not helpful... I'd like to see, I don't know? Images showing what's not matching?

Reply:
Personally I don't see a problem with the slight mismatch, I was just trying to match stable as much as possible since that seems to be the goal for classic

As a side note, it also happens to aspect ratio <= 4:3, but the difference there is also minor